File: 8-kyu-kata/grasshopper-messi-goals-function/python/grasshopper_messi_goals_function_test.py
# 8 kyu - Grasshopper - Messi goals function  [ ID: 55f73be6e12baaa5900000d4  (grasshopper-messi-goals-function) ]
# URL: https://www.codewars.com/kata/55f73be6e12baaa5900000d4
# Category: REFERENCE  |  Tags: FUNDAMENTALS
# ******************************************************************************
import codewars_test as test
from solution import goals


@test.describe("Fixed Tests")
def fixed_tests():
    @test.it("Basic Test Cases")
    def basic_test_cases():
        test.assert_equals(goals(0, 0, 0), 0)
        test.assert_equals(goals(5, 10, 2), 17)


@test.describe("Random Tests")
def random_tests():

    from random import randint

    for i in range(50):
        l = randint(0, 50)
        c = randint(0, 20)
        r = randint(0, 10)

        @test.it(f"testing for goals({l}, {c}, {r})")
        def test_case():
            test.assert_equals(goals(l, c, r), l + c + r)

File: kata-8-kyu/grasshopper-messi-goals-function/javascript/grasshopperMessiGoalsFunction.Test.js
//+ ====================================================================================================================
//+
//+ 8 kyu - Grasshopper - Messi goals function  [ ID: 55f73be6e12baaa5900000d4 ] (grasshopper-messi-goals-function)
//+ URL: https://www.codewars.com/kata/55f73be6e12baaa5900000d4
//+ Category: REFERENCE  |  Tags: FUNDAMENTALS
//+
//+ ====================================================================================================================

import { assert } from "vitest"
const { goals } = require("./grasshopperMessiGoalsFunction")

describe("goals test", function () {
  it("should be 0", function () {
    assert.strictEqual(goals(0, 0, 0), 0)
  })
  it("should be 58", function () {
    assert.strictEqual(goals(43, 10, 5), 58)
  })
  it("should work for random numbers", function () {
    let solution = (ll, cl, cdr) => ll + cl + cdr
    for (let i = 0; i < 100; i++) {
      let l = Math.floor(Math.random() * 50)
      let c = Math.floor(Math.random() * 20)
      let r = Math.floor(Math.random() * 10)
      assert.strictEqual(solution(l, c, r), goals(l, c, r))
    }
  })
})
